src/gtest/test_history.cpp

```diff
@@ -81,34 +81,36 @@ TEST(History, Smoky) {
     FakeCoinsViewDB fakeDB;
     CCoinsViewCache view(&fakeDB);
 
+    uint32_t epochId = 0;
+
     // Test initial value
-    EXPECT_EQ(view.GetHistoryLength(0), 0);
+    EXPECT_EQ(view.GetHistoryLength(epochId), 0);
 
-    view.PushHistoryNode(1, getLeafN(1));
+    view.PushHistoryNode(epochId, getLeafN(1));
```

Comment:
Where does epochId get incremented?

Reply:
It doesn't; this was a bug in the original test. This test is supposed to check for an empty tree, then append, and check the tree's length increases in size. But the test was using a different epoch ID for the empty tree check.